File: src/main/java/io/github/saneea/feature/Nowhere.java
package io.github.saneea.feature;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.security.NoSuchAlgorithmException;

import io.github.saneea.Feature;
import io.github.saneea.FeatureContext;

public class Nowhere implements Feature, Feature.In.Bin.Stream {

	private InputStream in;

	@Override
	public Meta meta(FeatureContext context) {
		return Meta.from("read all input data and do nothing");
	}

	@Override
	public void run(FeatureContext context) throws NoSuchAlgorithmException, IOException {
		in.transferTo(OutputStream.nullOutputStream());
	}

	@Override
	public void setIn(InputStream in) {
		this.in = in;
	}
}

File: src/main/kotlin/io/github/saneea/dvh/feature/Nowhere.kt
package io.github.saneea.dvh.feature

import io.github.saneea.dvh.Feature
import io.github.saneea.dvh.Feature.Meta
import java.io.InputStream
import java.io.OutputStream

class Nowhere :
    Feature,
    Feature.In.Bin.Stream {

    override lateinit var inBinStream: InputStream

    override val meta = Meta("read all input data and do nothing")

    override fun run() {
        inBinStream.transferTo(OutputStream.nullOutputStream())
    }
}